### PR TITLE
Add updated PHPStan in project as composer dependency

### PR DESCRIPTION
**Pull request by Quetzacoalt91**
| Questions         | Answers
| ------------------| -------------------------------------------------------
| Description?      | The version of PHPStan is quite old (https://packagist.org/packages/phpstan/phpstan#0.12.100). This PR updates it, merge all the different workflow to an existing  bash file then obviously fixes all the new errors reported by PHPStan.
| Type?             | improvement
| BC breaks?        | Nope
| Deprecations?     | Nope
| Fixed ticket?     | /
| Sponsor company   | @PrestaShopCorp
| How to test?      | Non regression tests should be run. Twig has been removed in this PR, so display of the configuration should be checked on different versions of PS (i.e 1.7.3, 1.7.6, 1.7.8, 8.0, 9.0)

### CODE REVIEW

**Review by M0rgan01 — DISMISSED** [review]

File: tests/composer.json [hunk 1]
```diff
@@ -0,0 +1,6 @@
+{
+    "require-dev": {
+        "phpstan/phpstan": "^1.11",
+        "prestashop/php-dev-tools": "dev-master"
```
Comment (Quetzacoalt91) [inline tests/composer.json:4]: Need to switch to a stable release

File: tests/unit/UpgradeContainerTest.php [hunk 2]
```diff
@@ -52,18 +52,21 @@ public function objectsToInstanciateProvider()
...
+            ['getFileLoader', PrestaShop\Module\AutoUpgrade\Xml\FileLoader::class],
             ['getLogger', PrestaShop\Module\AutoUpgrade\Log\LegacyLogger::class],
             ['getModuleAdapter', PrestaShop\Module\AutoUpgrade\UpgradeTools\ModuleAdapter::class],
             ['getState', \PrestaShop\Module\AutoUpgrade\State::class],
             ['getSymfonyAdapter', PrestaShop\Module\AutoUpgrade\UpgradeTools\SymfonyAdapter::class],
             ['getTranslationAdapter', \PrestaShop\Module\AutoUpgrade\UpgradeTools\Translation::class],
             ['getTranslator', \PrestaShop\Module\AutoUpgrade\UpgradeTools\Translator::class],
-            ['getTwig', \Twig\Environment::class],
+            // Cannot be run in the context of unit tests as the class would be loaded from PS dependencies
+            // ['getTwig', \Twig\Environment::class],
```
Comment (M0rgan01) [inline tests/unit/UpgradeContainerTest.php:69]: Why keep items in comments ?

File: tests/phpstan/phpstan-1.7.8.neon [hunk 3]
```diff
@@ -2,13 +2,16 @@ includes:
...
 		- '#Instantiated class PrestaShop\\PrestaShop\\Core\\Cldr\\Update not found.#'
-		- '#Call to method boot\(\) on an unknown class AdminKernel.#'
 		- '#Call to method getContainer\(\) on an unknown class AdminKernel.#'
-		- '#Return typehint of method PrestaShop\\Module\\AutoUpgrade\\UpgradeTools\\SymfonyAdapter::initKernel\(\) has invalid type AdminKernel.#'
+		- '#Method PrestaShop\\Module\\AutoUpgrade\\UpgradeTools\\SymfonyAdapter::initKernel\(\) has invalid return type AdminKernel.#'
 		- '#Instantiated class AdminKernel not found\.#'
+		-
+			identifier: booleanAnd.rightAlwaysTrue
+			path: ./../../classes/UpgradeSelfCheck.php
+			count: 2
```
Comment (M0rgan01) [inline tests/phpstan/phpstan-1.7.8.neon:17]: There are a few files that don't have a blank line at the end

File: tests/unit/UpgradeContainerTest.php (hunk 2 above)
Comment (Quetzacoalt91) [reply tests/unit/UpgradeContainerTest.php:69]: To provide the information that I did not forget to test this method, and trying it will trigger an error.

**Review by M0rgan01 — APPROVED** [review]

**Review by AureRita — APPROVED** [review]
Hi @Quetzacoalt91 

I tested your module from 1.7.1.0 to 1.7.3.0 as you can see : 

[recording(219).webm](https://github.com/PrestaShop/autoupgrade/assets/100699445/7d0372df-f2b4-4bc9-8475-5c088b8bb3d9)

and i have this error : 
image

But this one is not related to your PR.

Except that, That what I have : 

[recording(220).webm](https://github.com/PrestaShop/autoupgrade/assets/100699445/35d6b433-1703-464c-8c1f-26a045e2ac2c)

I tested it from : 
1.7.3.0 to 1.7.6.0 
1.7.6.0 to 1.7.8.9 
1.7.8.9 to 8.0.5 
8.0.0 to 9.0.0 

on all of them I do a rollback except for 8.0.0 to 9.0.0. 
I tested the autoupgrade module on 9.0.0 on it works 

Because the PR seems to works correctly, I think it's a QA :heavy_check_mark: 

Thank you !

**Pull request merged**